### DISCUSSION TITLE
Github Copilot caused VSCode to lag after long time of use

### DISCUSSION THREAD

**Discussion by feng-yu-healthbank** [community/community#52042] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I am on a trial plan of Github Copilot. Every time after a full day of work, in the late afternoon, performance of Copilot deteriorated heavily. Copilot would take half a minute to get a result after I type anything. VSCode's own completion would also take a long time to appear. Even code formatting would become tremendously slow to respond. VSCode UI also became sluggish and sometimes laggy.

Today I uninstalled Copilot and VSCode no longer lags. 

Could there be some sort of chronic memory leak in Copilot on Windows? It is nor usable to me with the current performance.

---

System

Edition	Windows 11 Pro
Version	22H2
Installed on	‎2023-‎03-‎19
OS build	22621.1485
Experience	Windows Feature Experience Pack 1000.22639.1000.0

VSCode

Version: 1.77.0 (system setup)
Commit: 7f329fe6c66b0f86ae1574c2911b681ad5a45d63
Date: 2023-03-29T10:02:16.981Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes

Environment

CPU: (24) x64 AMD Ryzen 9 5900X 12-Core Processor
Memory: 36.47 GB / 63.91 GB
GPU: AMD Radeon 7900XTX, 24GB VRAM
Binaries:
  Node: 18.2.0 - C:\Program Files\nodejs\node.EXE
  Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
  npm: 9.6.2 - C:\Program Files\nodejs\npm.CMD
  Watchman: 20230319.141037.0 - C:\ProgramData\chocolatey\bin\watchman.EXE
Languages:
    Java: 15.0.2 - C:\Program Files\Common Files\Oracle\Java\javapath\javac.EXE

Disk: Intel Optane 905p SSD. 
Network: 1000/50 Mbps fibre and wired ethernet.

**Comment by rookledookle** [comment]
Not sure why this was closed but am experiencing exactly the same thing - VS Code is unusable with Copilot for me

**Comment by MarianoMolina** [comment]
Same thing here. Been using it for a while, and been noticing this issue more and more, until now being forced to face it, and realizing it was copilot all along.

**Comment by epsilonion-liam** [comment]
probably due to the chat history i am thinking it it is a long chat history then click done and open another

**Comment by FelipeMartini** [comment]
Same here also in local work fine but with remote ssh extension lag a lot

**Comment by yanghoxom** [comment]
same for wsl, too lag after copilot run, need to reload window

**Comment by BlueRexPY** [comment]
Try `Ctrl + Shift + p` + `Chat: Clear All Workspace Chats`. Worked for me.

https://helgeklein.com/blog/fixing-vs-code-ui-unresponsiveness-caused-by-github-copilot-extension/

**Comment by YamiOdymel** [comment]
Related: Copilot Agent Mode is slowing down the app
https://github.com/microsoft/vscode/issues/268516
